test/e2e/operator_test.go

```diff
@@ -150,6 +152,99 @@ func TestRouterServiceInternalEndpoints(t *testing.T) {
 	}
 }
 
+func TestClusterProxyProtocol(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cm := &corev1.ConfigMap{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Namespace: util.ClusterConfigNamespace, Name: util.ClusterConfigName}, cm)
+	if err != nil {
+		t.Fatalf("failed to get clusterconfig: %v", err)
+	}
+	installConfig, err := util.UnmarshalInstallConfig(cm)
+	if err != nil {
+		t.Fatalf("failed to read clusterconfig: %v", err)
+	}
+
+	if installConfig.Platform.AWS == nil {
+		t.Skip("test skipped on non-aws platform")
+		return
+	}
+
+	ci := &ingressv1alpha1.ClusterIngress{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: "default"}, ci); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get default ClusterIngress: %v", err)
+	}
+
+	// Wait for the router deployment to exist.
+	deployment := &appsv1.Deployment{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-ingress", Name: fmt.Sprintf("router-%s", ci.Name)}, deployment); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get default router deployment: %v", err)
+	}
+
+	// Ensure proxy protocol is enabled on the router deployment.
+	proxyProtocolEnabled := false
+	for _, v := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if v.Name == "ROUTER_USE_PROXY_PROTOCOL" {
+			if val, err := strconv.ParseBool(v.Value); err == nil {
+				proxyProtocolEnabled = val
+				break
+			}
+		}
+	}
+
+	if !proxyProtocolEnabled {
+		t.Fatalf("expected router deployment to enable the PROXY protocol")
+	}
+
+	// Wait for the internal router service to exist.
+	internalService := &corev1.Service{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-ingress", Name: fmt.Sprintf("router-internal-%s", ci.Name)}, internalService); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get internal router service: %v", err)
+	}
+
+	// TODO: Wait for interal router service selector bug to be fixed.
```

Comment:
https://github.com/openshift/cluster-ingress-operator/pull/90 merged, so this TODO is stale, right?

Reply:
yeah but as per a comment from @ironcladlou previously I was debating on whether to do it here (or do it in origin as there's already a header test that checks for the `*Forwarded*` headers - that seems like a more complete blackbox e2e test).